model/flow/sealing_segment.go

```diff
@@ -0,0 +1,33 @@
+package flow
+
+// SealingSegment is the chain segment such that the head (greatest
+// height) is this snapshot's reference block and the tail (least height)
+// is the most recently sealed block as of this snapshot (ie. the block
+// referenced by LatestSeal). The segment is in ascending height order.
+type SealingSegment struct {
+	// Blocks the chain segment blocks
+	Blocks []*Block
+
+	// Due to decoupling of execution receipts it's possible that blocks from sealing segment will be referring
+	// execution results incorporated in blocks that aren't part of the segment.
+	// ExecutionResults will contain those results.
+	ExecutionResults ExecutionResultList
+}
+
+// AddBlock appends block to Blocks
+func (segment *SealingSegment) AddBlock(block *Block) {
+	segment.Blocks = append(segment.Blocks, block)
+}
+
+// AddExecutionResult adds result to ExecutionResults
+func (segment *SealingSegment) AddExecutionResult(result *ExecutionResult) {
+	segment.ExecutionResults = append(segment.ExecutionResults, result)
+}
```

Comment:
This is the expected behavior , ExecutionResults can be empty

Reply:
My point is that let's say a block A has a receipt that is referring to a unknown result to be queried. After queried that result, you will need to `AddExecutionResult` with that result, and then `AddBlock` with that block. But what happens if `AddExecutionResult` was called, but `AddBlock` wasn't? Or `AddBlock` was called, but `AddExecutionResult` wasn't? That would cause data inconsistency. And there is a possibility that this could happen, because the API design allows this. 

So what I'm suggesting to prevent this above from happening by combining `AddBlock` and `AddExecutionResult` ( or simply remove the `AddExecutionResult` ).

When adding a block, it should also find the missing result, and add it. In the end, there will be only 2 results:
either no block was added, or if a block is added, the missing result must have been added as well. The eliminates the cases where a block is added, but the result is not, or a result is added, but a block was not.